velox/type/Variant.cpp

```diff
@@ -236,11 +239,14 @@ std::string variant::toJson(const TypePtr& type) const {
       std::string b{};
       b += "[";
       bool first = true;
+      uint32_t idx = 0;
       for (auto& v : array) {
         if (!first) {
           b += ",";
         }
-        b += v.toJson();
+        VELOX_USER_CHECK_EQ(
```

Comment:
This check can be in the `toJson(const TypePtr& type)` method.
We can probably make the error message more descriptive.
```
VELOX_USER_CHECK_EQ(
            v.kind(), type->kind(), "variant TypeKind {} does not match input TypeKind {}", v.kind(), type->kind());
```

Reply:
We create `v` within the foreach loop for ROW and ARRAY TypeKinds. So we can't keep this check at toJson(). `v` would be inaccessible.

Reply:
It would be `this.kind()` instead of `v`.

Reply:
Changes the error message, but the check remains at same place.

Reply:
See my comment above.

Reply:
Fixed as above.